scripts/bin/auto-mount

```diff
@@ -1,17 +1,52 @@
-#! /bin/sh
-
-sleep 5
-
-$SNAP/usr/local/bin/udisksctl monitor | grep --line-buffered " Device:" | while read -r -- device; do
-
-  echo "$SNAP_NAME.auto-mount found ${device}"
-  if echo $device| grep -q "[0-9]$"; then
-    DEV="$(echo $device | sed 's/^Device: //')"
-  fi
-  if [ -n "$DEV" ]; then
-    RES=`$SNAP/usr/local/bin/udisksctl mount --no-user-interaction --block-device "$DEV"`
-    # process returned value into the mount point string
-    RES2=`echo $RES | cut -f4 -d\  | cut -f1 -d.`
-    echo $RES2 > $WATCHED
-  fi
-done
+#!/bin/sh
+
+. "$SNAP/bin/common"
+
+init_mount_dir() {
+    _arg_dir="$1"
+
+}
+
+watch_devices() {
+    inotifywait --quiet --monitor --event "moved_to" --event "delete" /dev/disk/by-uuid | \
+        while read -r _parent _action _uuid; do
+            case "$_action" in
+                MOVED_TO) try_mount "$_parent" "$_uuid" ;;
+                DELETE)   rm_mount  "$_uuid" ;;
+            esac
```

Comment:
We should have a third case, even if it's just `*) ;;`

Reply:
That's a good point. In theory, that should exhaust all the options, since we're only listening for `moved_to` and `delete` events, but who knows what other garbage could get thrown our way, especially if there's an error

Reply:
If we get an error, we just `panic() { panic | panic }`

Reply:
The "If I can't run, nobody can" method of error handling. I like it